### PR TITLE
bugfix(CFD): Facebook Pixel error handling

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

> **_Issue_:**
Facebook Pixel `custom_data` expects a string or number
According to Rudderstacks documentation of the [product searched](https://www.rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-ecommerce-events-specification/browsing/#products-searched) event, the query should be a string / Object
So, [here](https://github.com/rudderlabs/rudder-transformer/blob/1537d3ec5d932e603a011a33c0fc53a43623b33d/v0/destinations/facebook_pixel/transform.js#L496) query should be a string or number.
**_Fix:_**
We are handling it inside the transformer by throwing an appropriate error.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/facebook_pixel/transform.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const {
   isObject,
   getValidDynamicFormConfig
 } = require("../../util");
+const logger = require("../../../logger");
```
Comment (ItsSudip) [inline v0/destinations/facebook_pixel/transform.js:28]: Remove this, as it is not used.

File: v0/destinations/facebook_pixel/transform.js [hunk 2]
```diff
@@ -491,6 +492,10 @@ const responseBuilderSimple = (
           commonData.event_name = "Purchase";
           break;
         case "products searched":
+          const query = message.properties?.query;
+          if (query && typeof query !== "string" && typeof query !== "number") {
+            throw new CustomError("'query' should be in (string (or) number) format only", 400);
```
Comment (shrouti1507) [inline v0/destinations/facebook_pixel/transform.js:497]: According to Rudderstack spec, a Query should be a string or object. So an error message like `string or number`, would that not be misleading? WDYT @ItsSudip @utsabc

Comment (ItsSudip) [reply v0/destinations/facebook_pixel/transform.js:497]: Yeah, that is correct. But we are throwing this error as Facebook doesn't support `object`.

File: v0/destinations/facebook_pixel/transform.js (hunk 1 above)
Comment (rohithkaza) [reply v0/destinations/facebook_pixel/transform.js]: Removed

File: v0/destinations/facebook_pixel/transform.js [hunk 3]
```diff
@@ -493,8 +492,12 @@ const responseBuilderSimple = (
           break;
         case "products searched":
           const query = message.properties?.query;
-          if (query && typeof query !== "string" && typeof query !== "number") {
-            throw new CustomError("'query' should be in (string (or) number) format only", 400);
+          const validQueryType = ["string", "number", "boolean"];
```
Comment (shrouti1507) [inline v0/destinations/facebook_pixel/transform.js:495]: can you please add the reference url for valid query types in facebook pixel?

File: v0/destinations/facebook_pixel/transform.js (hunk 2 above)
Comment (shrouti1507) [reply v0/destinations/facebook_pixel/transform.js]: In that case comment there with your findings. 
like: 
```
/* custom_data field type ref : < link to the page>
    and also by behaviour it is found that it supports `boolean` and `number` too. As a result these two types of fields are allowed */
```

File: v0/destinations/facebook_pixel/transform.js (hunk 3 above)
Comment (rohithkaza) [reply v0/destinations/facebook_pixel/transform.js:495]: But, according to Facebook Pixel [here](https://developers.facebook.com/docs/meta-pixel/reference#:~:text=an%20exact%20value.-,search_string,-String), they stated it must be in `string` format. But as they are accepting `number` and `boolean` also. So, we included them too.

File: v0/destinations/facebook_pixel/transform.js (hunk 2 above)
Comment (rohithkaza) [reply v0/destinations/facebook_pixel/transform.js]: Done

File: v0/destinations/facebook_pixel/transform.js (hunk 3 above)
Comment (rohithkaza) [reply v0/destinations/facebook_pixel/transform.js:501]: Added reference URL

**Review by shrouti1507 — APPROVED** [review]
LGTM

**Review by utsabc — APPROVED** [review]

**Pull request merged**